crates/bones_ecs/src/resources.rs

```diff
@@ -304,6 +299,21 @@ impl<T: TypedEcsData> AtomicResource<T> {
         AtomicRefMut::map(borrow, |data| unsafe { &mut *data.cast::<T>() })
     }
 }
+impl<R: TypedEcsData> Deref for AtomicResource<R> {
+    type Target = R;
+
+    fn deref(&self) -> &Self::Target {
+        // SAFE: We know that the data pointer is valid for type UntypedResource.
+        unsafe { &*self.untyped.borrow().cast::<R>() }
+    }
+}
+
+impl<R: TypedEcsData> DerefMut for AtomicResource<R> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        // SAFE: We know that the data pointer is valid for type UntypedResource.
+        unsafe { &mut *self.untyped.borrow_mut().cast::<R>() }
+    }
+}
```

Comment:
OK, verified this test fails Miri:

```rust
    #[test]
    fn deref_atomicresource() {
        #[derive(TypeUlid, Clone, Debug)]
        #[ulid = "01GNDJJ42DY5GSP9PGXPGF65GE"]
        struct A(u32);

        let mut w = World::new();
        w.insert_resource(A(1));

        let mut r1 = w.resource::<A>();
        let r1mut = &mut *r1;

        let r2 = w.resource::<A>();
        let mut r2mut = r2.borrow_mut();

        r2mut.0 = 2;
        assert_eq!(r1mut.0, r2mut.0);
    }
```

It lets us get two mutable references to the resource at a time.

But, if we use `r1.borrow_mut()` instead of using the `DerefMut` implementation with `&mut *r1` instead of causing undefined behavior, the call of `r2.borrow_mut()` panics at runtime indicating that the resource has already been borrowed.